### PR TITLE
fix: modify githubactions template, add blank, support  push-main without PR 

### PR DESCRIPTION
**Pull request by lfbdev**
# Summary

There is no blank between "path:" and "coverage.out"

## Current Behavior

1. no blank
image

2. no PR when push directly 
image

## New Behavior

image

### Related Issues
#295 
#298

### CODE REVIEW

File: internal/pkg/plugin/githubactions/golang/create.go [hunk 1]
```diff
@@ -37,6 +37,10 @@ func Create(options map[string]interface{}) (map[string]interface{}, error) {
 		}
 	}
 
+	if err := ghClient.AddRepoSecret("GH_TOKEN", viper.GetString("github_token")); err != nil {
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/create.go:40]: This is not necessary

File: internal/pkg/plugin/githubactions/golang/delete.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func Delete(options map[string]interface{}) (bool, error) {
 
 	// if docker is enabled, delete repo secrets DOCKERHUB_USERNAME and DOCKERHUB_TOKEN
 	if opt.Docker != nil && opt.Docker.Enable {
-		for _, secret := range []string{"DOCKERHUB_USERNAME", "DOCKERHUB_TOKEN"} {
+		for _, secret := range []string{"DOCKERHUB_USERNAME", "DOCKERHUB_TOKEN", "GH_TOKEN"} {
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/delete.go:30]: Again, no need to create/delete GH_TOKEN

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Build
-      run: [[- if not .Build.Command]] go build ./...[[- else]][[.Build.Command]][[- end]]
+      run: [[ if not .Build.Command]] go build ./...[[- else]][[.Build.Command]][[- end]]
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:13]: [[ if not .Build.Command ]]

v.s.

[[ if not .Build.Command]]

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 4]
```diff
@@ -23,14 +23,14 @@ jobs:
       with:
         go-version: 1.17
     - name: Test
-      run: [[- if not .Test.Command]] go test ./...[[- else]][[.Test.Command]][[- end]] [[- if .Test.Coverage.Enable]] -race -covermode=atomic -coverprofile=[[- if not .Test.Coverage.Output]]coverage.out[[- else]][[.Test.Coverage.Output]][[- end]] [[- end]]
+      run: [[if not .Test.Command]] go test ./...[[- else]][[.Test.Command]][[- end]] [[- if .Test.Coverage.Enable]] -race -covermode=atomic -coverprofile=[[- if not .Test.Coverage.Output]]coverage.out[[- else]][[.Test.Coverage.Output]][[- end]] [[- end]]
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:26]: [[if not .Test.Command]]
the format is different from
[[ if not .Build.Command]]

space or not? one space at the beginning only? or one space at the beginning, one at the end? 

Make it consistent.

Also, please test both with coverate.output and without, and paste the screenshot here.

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 5]
```diff
@@ -23,14 +23,14 @@ jobs:
...
     [[- if .Test.Coverage.Enable]]
     - name: comment PR
       uses: machine-learning-apps/pr-comment@master
       env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:31]: Do not change the name. secrets.GITHUB_TOKEN already exists by default.

Same for all the followings.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 6]
```diff
@@ -23,14 +23,14 @@ jobs:
...
       with:
-        path: [[- if not .Test.Coverage.Output]]coverage.out[[- else]][[.Test.Coverage.Output]][[- end]]
+        path: [[if not .Test.Coverage.Output]] coverage.out[[- else]][[.Test.Coverage.Output]][[- end]]
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:33]: Again, [[if xxx] or [[ if xxx ]]

Keep it consistant

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 7]
```diff
@@ -81,7 +81,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Build
-      run: [[- if not .Build.Command]] go build ./...[[- else]][[.Build.Command]][[- end]]
+      run: [[if not .Build.Command]] go build ./...[[- else]][[.Build.Command]][[- end]]
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:84]: Same, please make sure all template rendering parts are consistent.

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 8]
```diff
@@ -94,14 +94,14 @@ jobs:
       with:
         go-version: 1.17
     - name: Test
-      run: [[- if not .Test.Command]] go test ./...[[- else]][[.Test.Command]][[- end]] [[- if .Test.Coverage.Enable]] -race -covermode=atomic -coverprofile=[[- if not .Test.Coverage.Output]]coverage.out[[- else]][[.Test.Coverage.Output]][[- end]] [[- end]]
+      run: [[if not .Test.Command]] go test ./...[[- else]][[.Test.Command]][[- end]] [[- if .Test.Coverage.Enable]] -race -covermode=atomic -coverprofile=[[- if not .Test.Coverage.Output]]coverage.out[[- else]][[.Test.Coverage.Output]][[- end]] [[- end]]
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:97]: Same, [[ if xxx]] [[if xxx]] [[ if xxx ]]

File: internal/pkg/plugin/githubactions/golang/templates.go [hunk 9]
```diff
@@ -94,14 +94,14 @@ jobs:
...
     [[- if .Test.Coverage.Enable]]
     - name: comment PR
       uses: machine-learning-apps/pr-comment@master
       env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
       with:
-        path: [[- if not .Test.Coverage.Output]]coverage.out[[- else]][[.Test.Coverage.Output]][[- end]]
+        path: [[if not .Test.Coverage.Output]] coverage.out[[- else]][[.Test.Coverage.Output]][[- end]]
```
Comment (IronCore864) [inline internal/pkg/plugin/githubactions/golang/templates.go:104]: Same, consistency is nice

**Review by daniel-hutao — APPROVED** [review]

**Review by IronCore864 — APPROVED** [review]

**Pull request merged**